index.html

```diff
@@ -122,6 +159,165 @@ <h2>Methods</h2>
+      initialized to the platform-specific initial set of RTP header extensions that
+      is offered on an initial CreateOffer. This list MUST include all header extensions
+      that are marked Mandatory to Implement in the underlying protocol. The direction
+      attribute for all extensions that are mandatory to use MUST be initialized to
+      "sendrecv".
+    </p>
+    <p class="note">The list of header extensions that MUST/SHOULD be supported is listed in
+      draft-ietf-rtcweb-rtp-usage, section 5.2. The "mid" extension is mandatory to use when
+      BUNDLE is in use.
+    </p>
+    <p>
+      Let [[\HeaderExtensionsNegotiated]] be an internal slot of the RTCRtpTransceiver,
+      initialized to an empty list.
+    </p>
+    <p>
+      In the algorithms for generating initial offers in JSEP section 5.1.1,
+      read "for each supported RTP header extension" as "For each RTP header extension
+      listed in [[\HeaderExtensionsToOffer]] where direction is not "stopped"".
+    </p>
+    <p class="note">
+      Since JSEP does not know about WebRTC internal slots, merging this change requires
+      more work on a JSEP revision.
+    </p>
+    <p>
+      In the algorithm to <a>set an SDP description</a>, add a bullet in the sections
+      for processing answer and pranswer (bullet 3.2.8.1.7 and 3.2.8.2.6):
+      <ul>
+        <li>Set the transceiver's [[\HeaderExtensionsNegotiated]] slot to the list of
+          RTP header extensins from the answer.
+        </li>
+      </ul>
+    </p>
+    <section>
+      <h2>Methods</h2>
+      <dl>
+        <dt>setOfferedRtpHeaderExtensions</dt>
+        <dd>
+          Execute the following steps:
+          <ol>
+            <li>Let <var>extlist</var> be the list of extensions.</li>
+            <li>
+              For each extension <var>ext</var> in <var>extlist</var>, do the following:
+              <ol>
+                <li>If <var>ext</var>'s "uri" value is not a supported extension, throw
+                  a NotSupported error.
+                </li>
+                <li>If <var>ext</var>'s "direction" value is not "sendrecv", and "uri"
+                  indicates a mandatory-to-use attribute, throw an IllegalArgument error.
+                </li>
+                <li>if <var>ext</var>'s "direction" value is "stopped", and "uri" indicates
+                  a mandatory-to-implement extension, throw an IllegalArgument error.
+                </li>
+              </ol>
+            </li>
+            <li>
+              For each mandatory-to-implement extension <var>mext</var>, if <var>mext</var>
+<<<<<<< HEAD
+              is not present in <extlist>, throw an IllegalArgument error.
+=======
+              is not present in <var>extlist</var>, throw an IllegalArgument error.
+>>>>>>> 441d821224ddbc98724c985e9639cffe2e20344e
+            </li>
+            <li>
+              Set the [[\HeaderExtensionsToOffer]] to <var>extlist</var>.
+            </li>
```

Comment:
"JSEP 5.2.2 Subsequent Offers" says:

> In addition, for each existing, non-recycled, non-rejected m= section in the new offer, the following adjustments are made based on the contents of the corresponding m= section in the current local or remote description, as appropriate:
> ...
> - The RTP header extensions MUST only include those that are present in the most recent answer.

If we want to override this behavior we must be clear that this is intentional. If we don't want to override this behavior, we should probably throw an exception if you attempt to use this API after [[\HeaderExtensionsNegotiated]] has a value.